mokapot/plugins/__init__.py

```diff
@@ -0,0 +1,43 @@
+from __future__ import annotations
+
+import sys
+import logging
+
+if sys.version_info >= (3, 10):
+    from importlib.metadata import entry_points
+else:
+    from importlib_metadata import entry_points
+
+from argparse import _ArgumentGroup
+from typing import Any
+
+from mokapot.config import Config
+from mokapot.model import Model
+
+LOGGER = logging.getLogger(__name__)
+
+
+def get_plugins() -> dict[str, Any]:
+    """Return a dict of all installed Plugins as {name: EntryPoint}."""
+
+    plugins = entry_points(group="mokapot.plugins")
+
+    pluginmap = {}
+    for plugin in plugins:
+        pluginmap[plugin.name] = plugin
+
+    for k, v in pluginmap.items():
+        LOGGER.debug(f"loading {k}")
+        pluginmap[k] = v.load()
+    return pluginmap
+
+
+class BasePlugin:
+    def add_arguments(parser: _ArgumentGroup) -> None:
```

Comment:
**question:** Is there a way to avoid using a private class in the type hint here?

Reply:
I do not beleive there is ... TECHNICALLY we could use ArgumentParser but it would not have all the methods implemented (although it is close enough for most cases ... https://github.com/python/typeshed/blob/master/stdlib/argparse.pyi)